src/material/slide-toggle/testing/slide-toggle-harness.ts

```diff
@@ -6,32 +6,38 @@
  * found in the LICENSE file at https://angular.io/license
  */
 
-import {ComponentHarness, HarnessPredicate} from '@angular/cdk/testing';
+import {
+  AsyncFactoryFn,
+  ComponentHarness,
+  HarnessPredicate,
+  TestElement,
+} from '@angular/cdk/testing';
 import {coerceBooleanProperty} from '@angular/cdk/coercion';
 import {SlideToggleHarnessFilters} from './slide-toggle-harness-filters';
 
 export abstract class _MatSlideToggleHarnessBase extends ComponentHarness {
   private _label = this.locatorFor('label');
-  protected _input = this.locatorFor('input');
+  protected abstract _nativeElement: AsyncFactoryFn<TestElement>;
 
   /** Toggle the checked state of the slide-toggle. */
   abstract toggle(): Promise<void>;
 
   /** Whether the slide-toggle is checked. */
   async isChecked(): Promise<boolean> {
-    const checked = (await this._input()).getProperty<boolean>('checked');
+    // Use aria-checked since it applies for both MDC and non-MDC.
+    const checked = (await this._nativeElement()).getAttribute('aria-checked');
```

Comment:
Can we change this back for the non-MDC slide toggle and just override it for the MDC-based one? There seem to be some issues with change detection that cause a number of tests to fail with this change

Reply:
Fixed.